test/files/jvm/future-spec/FutureTests.scala

```diff
@@ -817,6 +817,24 @@ class FutureTests extends MinimalScalaTest {
       Await.ready(f, defaultTimeout).value.get.toString mustBe expected.toString
     }
 
+    "should not deadlock with a barrage of blocking {} (ticket 10484)" in {
+      val time = 2000
+      def spawn(nTasks: Int, name: String, blocking: Boolean): Unit = {
+        def log(str: String): Unit = println(s"${name}(${Thread.currentThread().getName}): $str")
+        val all = Future.sequence(Vector.range(1, nTasks + 1).map {
+          result => Future {
+            if (blocking) scala.concurrent.blocking { Thread.sleep(time) }
+            else Thread.sleep(time)
+            //log(s"$result - sleep was: $time - ${java.util.Calendar.getInstance().getTime()}")
+            result
+          }
+        })
+        //log("waiting")
+        Await.result(all, Duration(s"${10 + (time.toLong * nTasks)}s"))
```

Comment:
Duration for this waiting should be smaller. 
- For the blocking case, it is: `(nTasks / (scala.concurrent.context.maxExtraThreads + scala.concurrent.context.maxExtraThreads) + 1) * sleepTime + minorAdditionForTestStability`. (minorAdditionForTestStability can be well below 10 seconds, eg. 1-2 should be heaps)
- For the non-blocking case it is: `(nTasks / scala.concurrent.context.maxThreads + 1) * sleepTime + minorAdditionForTestStability`.

Reply:
@avkonst It is as it is because it doesn't look at number of available threads. Can have a look at that later, but it is a bit delicate because we cannot rely on specific pool behavior (since it uses the JDK FJP)